geolocation/ios/Plugin/GeolocationPlugin.swift

```diff
@@ -0,0 +1,200 @@
+import Foundation
+import CoreLocation
+import UIKit
+import Capacitor
+
+public struct GeolocationCoords {
+    public var latitude: Double
+    public var longitude: Double
+    public init(latitude: Double, longitude: Double) {
+        self.latitude = latitude
+        self.longitude = longitude
+    }
+}
+
+@objc(GeolocationPlugin)
+public class GeolocationPlugin: CAPPlugin, CLLocationManagerDelegate {
+
+    enum CallType {
+        case permissions
+        case singleUpdate
+        case watch
+    }
+
+    var locationManager = CLLocationManager()
+    private var callQueue: [String: CallType] = [:]
+
+    @objc func getCurrentPosition(_ call: CAPPluginCall) {
+        call.save()
+        callQueue[call.callbackId] = .singleUpdate
+
+        DispatchQueue.main.async {
+            self.locationManager.delegate = self
+            self.locationManager.requestWhenInUseAuthorization()
+
+            if call.getBool("enableHighAccuracy", false) == true {
+                self.locationManager.desiredAccuracy = kCLLocationAccuracyBest
+            } else {
+                self.locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers
+            }
+
+            self.locationManager.requestLocation()
+        }
+    }
+
+    @objc func watchPosition(_ call: CAPPluginCall) {
+        call.save()
+        callQueue[call.callbackId] = .watch
+
+        DispatchQueue.main.async {
+            self.locationManager.delegate = self
+            self.locationManager.requestWhenInUseAuthorization()
```

Comment:
watchPosition should check the current permission status first. According [to the documentation for `requestWhenInUseAuthorization`](https://developer.apple.com/documentation/corelocation/cllocationmanager/1620562-requestwheninuseauthorization), the method does not trigger any callbacks if the status is not `notDetermined`. So if it has been denied, no callbacks will happen and it will just quietly do nothing.